src/main/scala/nl.knaw.dans.easy.deposit/servlets/DepositServlet.scala

```diff
@@ -159,49 +157,28 @@ class DepositServlet(app: EasyDepositApiApp)
       ).getOrRecoverResponse(respond)
   }
 
-  private def uploadFileItem(uuid: UUID, path: Path, uploadItem: FileItem): Try[(Boolean, FileItem)] = {
+  private def uploadFileItem(uuid: UUID, path: Path, uploadItem: FileItem): Try[FileItem] = {
+    managed(uploadItem.getInputStream).apply[Try[_]] {
```

Comment:
The types don't really line up, do they? `unzipDepositFile` has `Try[Unit]` while the second case has `Try[FileItem]`. It is therefore probably not correct to say in the return type that it has type `Try[FileItem]`... Any clue how you could fix that?

Reply:
Here I only need `Try[Unit]` for both cases. The result of `writeDepositFile` is required for another call:

https://github.com/DANS-KNAW/easy-deposit-api/blob/6a474d3d7683644d8890a9fc12a540840fa23c74/src/main/scala/nl.knaw.dans.easy.deposit/servlets/DepositServlet.scala#L145

The logic is subject to change anyhow because of #66 : one zip only or none. The current logic unzips any zip.

Reply:
Mmm, maybe I'm a bit confused here; are you saying that `writeDepositFile` is used twice (perfectly OK!), but that in this instance you only need the `Unit` return type? That isn't reflected by the return type of `uploadFileItem`. Anyway, I let you play with it. Just noticed this type conflict 😃